dotnet/src/Extensions/TemplateEngine.Handlebars/HandlebarsPromptTemplate.cs

```diff
@@ -25,7 +24,7 @@ public HandlebarsPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactor
     }
 
     /// <inheritdoc/>
-    public async Task<string> RenderAsync(Kernel kernel, ContextVariables variables, CancellationToken cancellationToken = default)
+    public async Task<string> RenderAsync(Kernel kernel, IDictionary<string, string> arguments, CancellationToken cancellationToken = default)
```

Comment:
Can you pass in `KernelFunctionArguments` because we want to access the execution settings also and if anything new gets added we may want the template rendering to have access to that.

Reply:
Yep, I'll do that shortly after this PR in scope of this issue - https://github.com/microsoft/semantic-kernel/issues/3761